src/render/resolved_dependencies.rs

```diff
@@ -594,6 +596,31 @@ pub async fn resolve_dependencies(
         None
     };
 
+    let host_env = if merge_build_host {
```

Comment:
I think this is the wrong place. The merging should happen of the `matchspecs`, and before the environment is resolved. 

E.g. if you have

```
build:
  - compiler
host:
  - bla
```

It should just insert `compiler` into the `host` env (before resolving & installing) and then resolve everything at once into the same environment. We shouldn't have to touch the `ResolvedDependencies` as they should be fine (at that stage).

```
    let match_specs = specs.iter().map(|s| s.spec().clone()).collect::<Vec<_>>();
    // insert build match specs here!
    let host_env = if !match_specs.is_empty() {
```

Reply:
oh, that makes sense. Yeah pretty easy to change.